### DISCUSSION TITLE
(De)serialization in Godot

### DISCUSSION THREAD

**Discussion by MathiasBaumgartinger** [godotengine/godot-proposals#10643] [Scripting]
I am working on a project, where certain layers can be configured/stored in a json file. Serialization is somewhat straight forward. To deserialize those objects, I want to use `Object.get_property_list()` to get all members of the corresponding class. The initial idea was to handle the deserialization (i.e. a `Resource` is stored as path to the resource) via the class_name of the object. 

I have a few custom classes which would require more sophisticated (de)serialization. For instance a dictionary of `String` to a custom `Resource`. I would need the name of the custom class to handle this specific case. Sadly,  `Object.get_property_list()` only returns the native base class.

For example:

```
class CustomClass extends RefCounted:
   func _init(): 
     ...
``` 

would show "RefCounted" in the property list. In some forum post I read that this was intended behaviour. However, personally I feel this is somewhat of a limitation.

Does anyone have an idea on how I might handle this differently?

**Comment by FireCatMagic** [comment]
Have you tried Object.get_script().get_script_property_list()?

**Reply by MathiasBaumgartinger** [reply]
I found that too. The problem with that however, is that it requires me to have an ***Object*** of the _Class_. As I am deserializing a class which has members of another defined class, I have no object for the child classes.

I thought about just initializing an empty class so I could use `get_script_property_list()`. That seems like a hacky and unclean solution to me though.
